blocks/api/utils.js

```diff
@@ -66,7 +65,7 @@ export function normalizeIconObject( icon ) {
 	if ( ! icon ) {
 		return { src: 'block-default' };
 	}
-	if ( isString( icon ) || isFunction( icon ) || icon instanceof Component ) {
+	if ( ! icon.src ) {
```

Comment:
My concern with this is the lack of validation we provide. In contrast, `registerBlockType` does so much for block authors, but here we don't look at `icon` at all.

As also seen in `BlockIcon`, we have three types for WPIcon:
- string
- element
- component taking zero arguments 

Since there are no variable arguments at play, we can confidently test for the three cases:
- `typeof icon === 'string'`
- `wp.element.isValidElement( icon )`
- `wp.element.isValidElement( wp.element.createElement( icon ) )`

If all fail, log an error during `registerBlockType` and bail, as for any other validation error in that function.

This requires exposing React's `isValidElement` in `wp.element`.

Reply:
Hi @mcsf, thank you for the detailed feedback. I was not able to use 
wp.element.isValidElement( wp.element.createElement( icon ) ), because if we pass something invalid to wp.element.createElement, the function throws an error, so we need to check if we are in the presence of a function or a Component.